docs/src/further_topics/ugrid/data_model.rst

```diff
@@ -0,0 +1,565 @@
+          listed in "anti-clockwise order from above".
+        * Connectivities also exist to connect the higher dimensional elements,
+          e.g. ``face_edge_connectivity``. These are optional conveniences to
+          speed up certain operations and will not be discussed here.
+
+.. important::
+
+        **Meshes are unstructured**. The mesh elements - represented in the
+        coordinate and connectivity arrays detailed above - are enumerated
+        along a single **unstructured dimension**. An element's position along
+        this dimension has nothing to do with its spatial position.
+
+A data variable associated with a mesh has a **location** of either ``node``,
+``edge``, ``face`` or ``volume``. The data is stored in a 1D array with one
+datum per element, matched to its element by matching the datum index with the
+coordinate or connectivity index along the **unstructured dimension**. So for
+an example data array called ``foo``:
+``foo[3]`` would be at position ``(x[3], y[3])`` if it were node-located, or at
+``faces[3]`` if it were face-located. :numref:`data_ugrid_mesh` visualises an
+example of what is described above.
+
+..  _data_ugrid_mesh:
+..  figure:: images/data_ugrid_mesh.svg
+    :alt: Diagram of how data is represented on an unstructured mesh
+    :align: right
+    :width: 1280
+
+    Data on an unstructured mesh
+
+    1D coordinate arrays (pink circles) describe node positions in space (pink
+    crosses). A 2D connectivity array (blue circles) describes faces by
+    connecting four nodes - by referencing their indices - into a face outline
+    (blue outlines on the map). Data from the 1D data array (orange circles)
+    can be assigned to these node locations (orange diamonds) or face locations
+    (orange quads) by matching the indices in the data array to the indices in
+    the coordinate arrays (for nodes) or connectivity array (for faces). See
+    the black outlined shapes as examples of index matching in action, and the
+    black stippled shapes to demonstrate that relative array position confers
+    no relative spatial information.
+
+----
+
+The mesh model also supports edges/faces/volumes having associated 'centre'
+coordinates - to allow point data to be assigned to these elements. 'Centre' is
+just a convenience term - the points can exist anywhere within their respective
+elements. See :numref:`ugrid_element_centres` for a visualised example.
+
+..  _ugrid_element_centres:
+..  figure:: images/ugrid_element_centres.svg
+    :alt: Diagram demonstrating mesh face-centred data.
+    :align: right
+    :width: 1280
+
+    Data can be assigned to mesh edge/face/volume 'centres'
+
+    1D *node* coordinate arrays (pink circles) describe node positions in
+    space (pink crosses). A 2D connectivity array (blue circles) describes
+    faces by connecting four nodes into a face outline (blue outlines on the
+    map). Further 1D *face* coordinate arrays (pink circles) describe a
```

Comment:
Sorry to be late to the table here, but I think it seems unnecessarily confusing here to use the same colour for face coordinates as you do for node coordinates.  Is there a particular logic to the colour coding, which I've missed, which forces that choice ?

Reply:
1. Using more than 3 colours from the [IBM colour-blind palette](https://davidmathlogic.com/colorblind/#%23648FFF-%23785EF0-%23DC267F-%23FE6100-%23FFB000) puts them too close together under some types of colour-blindness.
2. Although number 1 was my original reason, I also like that they're both coordinates and therefore are more similar to each other than the other concepts in the diagrams.

Reply:
There are perhaps other palettes that allow four distinct colours but I wanted to pick one that was well recognised.